code/backend/batch/utilities/helpers/AzureSearchHelper.py

```diff
@@ -70,25 +122,67 @@ def get_vector_store(self):
-            azure_search_key=(
-                self.env_helper.AZURE_SEARCH_KEY
-                if self.env_helper.AZURE_AUTH_TYPE == "keys"
-                else None
-            ),
-            index_name=self.env_helper.AZURE_SEARCH_INDEX,
-            embedding_function=self.llm_helper.get_embedding_model().embed_query,
+        index = SearchIndex(
+            name=self.env_helper.AZURE_SEARCH_INDEX,
             fields=fields,
-            search_type=(
-                "semantic_hybrid"
-                if self.env_helper.AZURE_SEARCH_USE_SEMANTIC_SEARCH
-                else "hybrid"
+            semantic_search=(
+                SemanticSearch(
+                    configurations=[
+                        SemanticConfiguration(
+                            name=self.env_helper.AZURE_SEARCH_SEMANTIC_SEARCH_CONFIG,
+                            prioritized_fields=SemanticPrioritizedFields(
+                                title_field=None,
+                                content_fields=[SemanticField(field_name="content")],
+                            ),
+                        )
+                    ]
+                )
+            ),
+            vector_search=VectorSearch(
+                algorithms=[
+                    HnswAlgorithmConfiguration(
+                        name="default",
+                        parameters=HnswParameters(
+                            metric=VectorSearchAlgorithmMetric.COSINE
+                        ),
+                        kind=VectorSearchAlgorithmKind.HNSW,
+                    ),
+                    ExhaustiveKnnAlgorithmConfiguration(
+                        name="default_exhaustive_knn",
+                        kind=VectorSearchAlgorithmKind.EXHAUSTIVE_KNN,
+                        parameters=ExhaustiveKnnParameters(
+                            metric=VectorSearchAlgorithmMetric.COSINE
+                        ),
+                    ),
+                ],
+                profiles=[
+                    VectorSearchProfile(
+                        name="myHnswProfile",
+                        algorithm_configuration_name="default",
+                    ),
+                    VectorSearchProfile(
+                        name="myExhaustiveKnnProfile",
+                        algorithm_configuration_name="default_exhaustive_knn",
+                    ),
+                ],
             ),
-            semantic_configuration_name=self.env_helper.AZURE_SEARCH_SEMANTIC_SEARCH_CONFIG,
-            user_agent="langchain chatwithyourdata-sa",
         )
 
+        if self._index_not_exists(self.env_helper.AZURE_SEARCH_INDEX):
+            try:
+                logger.info(
+                    f"Creating or updating index {self.env_helper.AZURE_SEARCH_INDEX}"
+                )
+                self.search_index_client.create_index(index)
+            except Exception as e:
+                logger.exception("Error Creating index")
+                raise e
```

Comment:
suggestion: generally, logging an error and throwing an exception is considered an anti-pattern.
Since, throwing an exception will inform the user of this function that something has gone wrong, logging an `error` doesn't provide an additional value, instead `logger.info` can be used to log meaningful information for the debugging activity. Also, if there are log-scanners (`generally`, there should be) to raise alarms/tickets on error log count, then we get notified of 1 error 2 times - once through the thrown exception and second from log scanning.

Reply:
also, do we want to log `which index?`

Reply:
Yea this is a good point, I think this is probably left over from testing. I'll remove as I think the log before we create, and the contents of the exception should be sufficient.